compute_endpoint/globus_compute_endpoint/endpoint/endpoint.py

```diff
@@ -655,62 +656,80 @@ def stop_endpoint(
 
     @staticmethod
     def delete_endpoint(
-        endpoint_dir: pathlib.Path, endpoint_config: Config | None, force: bool = False
+        ep_dir: pathlib.Path | None,
+        ep_config: Config | None = None,
+        force: bool = False,
+        ep_uuid: str | None = None,
     ):
-        ep_name = endpoint_dir.name
+        ep_name = None
 
-        if not endpoint_dir.exists():
-            log.warning(f"Endpoint <{ep_name}> does not exist")
+        if ep_dir:
+            ep_name = str(ep_dir)
+        elif ep_uuid:
+            ep_name = ep_uuid
+        else:
+            log.warning("Name or UUID is needed to delete an Endpoint")
             exit(-1)
 
-        endpoint_id = Endpoint.get_endpoint_id(endpoint_dir)
-        if endpoint_id is None:
-            log.warning(
-                f"Configuration for endpoint <{ep_name}> could not be found, it "
-                "might not have been initialized locally"
-            )
-            if not force:
+        # If we have the UUID, do the online status check/deletion first
+        if ep_uuid:
+            try:
+                uuid.UUID(ep_uuid)
+            except (ValueError, TypeError):
+                log.warning(f"{ep_uuid} is not a valid endpoint UUID")
```

Comment:
Don't we already perform this check in `get_ep_dir_by_name_or_uuid()`?

Reply:
Yes thoughj that comes through the Click CLI.  I was imaging that other methods (or not necessarily invoking through the `delete` command) might use this method, so left it from the earlier iteration.  I can remove it if you think that's unlikely to be added in the future.

Reply:
Since this method is only intended for internal consumption, I think we should rely on the consumer (e.g., the CLI) to provide a valid UUID.

Reply:
Removed